kedro/framework/project/__init__.py

```diff
@@ -283,7 +315,28 @@ def find_pipelines() -> Dict[str, Pipeline]:
             function, the ``create_pipeline`` function does not return a
             ``Pipeline`` object, or if the module import fails up front.
     """
-    pipelines_dict = {"__default__": pipeline([])}
+    pipeline_module = None
+
+    # Handle the simplified project structure found in several starters.
+    pipeline_module_name = f"{PACKAGE_NAME}.pipeline"
+    try:
+        pipeline_module = importlib.import_module(pipeline_module_name)
+    except ModuleNotFoundError as exc:
+        if str(exc) != f"No module named '{pipeline_module_name}'":
```

Comment:
I'm not a big fan of string matching exception messages tbh - it feels kind of hacky. Or is this a well-accepted pattern? Can't we just see if the file `{PACKAGE_NAME}/pipeline.py` exists (or maybe there's an `importlib` way to check without actually doing the import? like how you do `importlib_resources.files`) and then only try to `import_module` on it if it does?

Reply:
Possible alternative: maybe put this `if` into the general `except` clause and remove this `except` block? It just doesn't feel quite right to me as it stands.

Reply:
> like how you do `importlib_resources.files`) and then only try to `import_module` on it if it does?

`importlib_resources.files` actually does `importlib.import_module` under the hood to make sure that the package passed to `files` is actually importable; I was actually about to go add another instance of string matching there! Let me see...

Reply:
Went ahead and merged the `except` clauses, but going to keep the string matching, as discussed. Trying to load based on folders seems overly complicated, because it would involve calling `bootstrap_project`.